aten/src/ATen/native/cpu/Reduce.h

```diff
@@ -104,15 +97,33 @@ void binary_kernel_reduce(TensorIterator& iter, ops_t ops, init_t init) {
           in += stride;
         }
       }, {begin, end});
-    });
-    acc_t acc = init;
-    for (int i = 0; i < max_threads; ++i) {
-      if (buffer[i]) {
-        acc = ops.combine(acc, *buffer[i]);
+    };
+    optional<acc_t> total_acc;
+    auto numel = sub_iter.numel();
+    if (numel < at::internal::GRAIN_SIZE || at::get_max_threads() == 1 || at::in_parallel_region()) {
+      reduction_body(total_acc, 0, numel);
+    } else {
+      int max_threads = at::get_max_threads();
+      AT_ASSERT(max_threads > 0);
+      std::vector<optional<acc_t>> buffer((unsigned)max_threads, optional<acc_t> {});
```

Comment:
I think you can do `nullopt` is a little more explicit than `optional<acc_t> {}`, but this is fine too.